src/editors/IED.ts

```diff
@@ -129,12 +129,22 @@ export default class IedPlugin extends LitElement {
           this.selectedIEDs = [iedName];
         }
       }
-      this.selectedLNClasses = this.lnClassList.map(
-        lnClassInfo => lnClassInfo[0]
-      );
     }
   }
 
+  private calcSelectedLNClasses(): string[] {
+    const somethingSelected = this.selectedLNClasses.length > 0;
+    const lnClasses = this.lnClassList.map( lnClassInfo => lnClassInfo[0] );
+
+    let selectedLNClasses = lnClasses;
+
+    if(somethingSelected){
+      selectedLNClasses = lnClasses.filter( lnClass => this.selectedLNClasses.includes(lnClass));
+    }
+
+    return selectedLNClasses;
+  }
+
```

Comment:
This is the main logic.
I've went for readability. You could argue, that we could wait with the mapping until nothing is selected, but I don't think this so such a big performance problem and it would be the code less readable.

Reply:
I like it!